### PR TITLE
feat: Add EC2 Instance Statuses

### PR DESCRIPTION
**Pull request by dakotalillie**
I'm submitting this PR on behalf on @mlozoya2, who has not yet been authorized to submit PRs to CloudQuery himself.

### CODE REVIEW

File: resources/services/ec2/instance_statuses.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+		IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
+		DeleteFilter: client.DeleteAccountRegionFilter,
+		Columns: []schema.Column{
+			{
+				Name:        "account_id",
+				Description: "The AWS Account ID of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSAccount,
+			},
+			{
```
Comment (bbernays) [inline resources/services/ec2/instance_statuses.go:28]: I would add an `arn` column just like in the `aws_ec2_instances`

File: resources/services/ec2/instance_statuses.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	"github.com/cloudquery/cq-provider-aws/client"
+
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func Ec2InstanceStatuses() *schema.Table {
+	return &schema.Table{
+		Name:         "aws_ec2_instance_statuses",
+		Description:  "Describes the status of an instance.",
+		Resolver:     fetchEc2InstanceStatuses,
```
Comment (bbernays) [inline resources/services/ec2/instance_statuses.go:17]: You should have a PK on each table. For this table you can use the ARN field (see comment below)

File: resources/services/ec2/instance_statuses.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+				Resolver:    schema.PathResolver("InstanceState.Code"),
+			},
+			{
+				Name:        "instance_state_name",
+				Description: "The current state of the instance.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("InstanceState.Name"),
+			},
+			{
+				Name:        "instance_status_details",
```
Comment (bbernays) [inline resources/services/ec2/instance_statuses.go:57]: I would remove the `instance_status_` prefix from column names

File: resources/services/ec2/instance_statuses.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+		res <- output.InstanceStatuses
+		if aws.ToString(output.NextToken) == "" {
+			break
+		}
+		config.NextToken = output.NextToken
+	}
+	return nil
+}
+func fetchEc2InstanceStatusEvents(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- interface{}) error {
+	r := parent.Item.(types.InstanceStatus)
```
Comment (bbernays) [inline resources/services/ec2/instance_statuses.go:160]: I would check the casting was successful:

```suggestion
	r, ok := parent.Item.(types.InstanceStatus)
	if !ok {
		return fmt.Errorf("not ec2 instance status")
	}
	
```

File: resources/services/ec2/instance_statuses.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "system_status_details",
+				Description: "The system instance health or application instance health.",
+				Type:        schema.TypeJSON,
+				Resolver:    schema.PathResolver("SystemStatus.Details"),
+			},
+			{
+				Name:        "system_status_status",
```
Comment (bbernays) [inline resources/services/ec2/instance_statuses.go:80]:
```suggestion
				Name:        "system_status",
```

**Review by bbernays — COMMENTED** [review]
Thank you very much for this PR! A few small comments, but overall looks great!!

**Comment by mlozoya2** [comment]
@bbernays Thanks for the review! We pushed some changes to address your comments. Let me know if there is anything else we need for this.

**Review by roneli — APPROVED** [review]
LGTM

File: resources/services/ec2/instance_statuses.go [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+				Description: "The AWS Region of the resource.",
+				Type:        schema.TypeString,
+				Resolver:    client.ResolveAWSRegion,
+			},
+			{
+				Name:        "arn",
+				Description: "The Amazon Resource Name (ARN) for the resource.",
+				Type:        schema.TypeString,
+				Resolver: client.ResolveARN(client.EC2Service, func(resource *schema.Resource) ([]string, error) {
+					return []string{"instance-status", *resource.Item.(types.InstanceStatus).InstanceId}, nil
```
Comment (bbernays) [inline resources/services/ec2/instance_statuses.go:42]: i think the resource should be just `instance` rather than `instance-status`

**Review by bbernays — CHANGES_REQUESTED** [review]

**Review by bbernays — APPROVED** [review]

**Pull request merged**